### PR TITLE
added iris version to index title

### PR DESCRIPTION
**Pull request by tkknight**
### Description

Add the Iris **version** to the main (index.rst) landing page.  Uses the already existing `version` variable.

This will only appear on the index, not on all pages.

Related to #3992.

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: docs/src/index.rst [hunk 1]
```diff
@@ -1,7 +1,7 @@
 .. _iris_docs:
 
-Iris Documentation
-==================
+Iris |version|
+========================
```
Comment (tkknight) [inline docs/src/index.rst:4]: As we are using a variable for the text of the heading, the `====` line needs to be the same length or more else there will be a build warning (and thus error).

**Comment by tkknight** [comment]
Built docs viewable for the lifetime of this PR at https://tkknight-iris-test-doc.readthedocs.io/en/latest/

**Review by bjlittle — APPROVED** [review]
Nicely done :+1:

**Comment by rcomer** [comment]
Thanks guys ☺️

**Pull request merged**